receiver/sshcheckreceiver/config.go

```diff
@@ -0,0 +1,68 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sshcheckreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver"
+
+import (
+	"errors"
+	"regexp"
+
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver/internal/configssh"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver/internal/metadata"
+)
+
+var rx = regexp.MustCompile(`^(([a-zA-Z]{1})|([a-zA-Z]{1}[a-zA-Z]{1})|([a-zA-Z]{1}[0-9]{1})|([0-9]{1}[a-zA-Z]{1})|([a-zA-Z0-9][a-zA-Z0-9-_]{1,61}[a-zA-Z0-9])).([a-zA-Z]{2,6}|[a-zA-Z0-9-]{2,30}.[a-zA-Z]{2,3})(:[0-9]{1,5})?$`)
+
+// Predefined error responses for configuration validation failures
+var (
+	errMissingEndpoint           = errors.New(`"endpoint" not specified in config`)
+	errInvalidEndpoint           = errors.New(`"endpoint" is invalid`)
+	errMissingUsername           = errors.New(`"username" not specified in config`)
+	errMissingPasswordAndKeyFile = errors.New(`either "password" or "keyfile" is required`)
+
+	errConfigNotSSHCheck  = errors.New("config was not a SSH check receiver config")
+	errWindowsUnsupported = errors.New(typeStr + " is unspported on Windows.")
+)
+
+type Config struct {
+	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
+	configssh.SSHClientSettings             `mapstructure:",squash"`
+	Metrics                                 metadata.MetricsSettings `mapstructure:"metrics"`
+}
+
+// SFTPEnabled tells whether any SFTP metrics are Enabled in MetricsSettings.
+func (c Config) SFTPEnabled() bool {
+	return (c.Metrics.SshcheckSftpDuration.Enabled || c.Metrics.SshcheckSftpStatus.Enabled)
+}
+
+func (c Config) Validate() (err error) {
+	if c.Endpoint == "" {
+		err = multierr.Append(err, errMissingEndpoint)
+	} else if !rx.MatchString(c.Endpoint) {
```

Comment:
Could you use https://pkg.go.dev/net#SplitHostPort here?

Reply:
Seems right. Fixed in 95874db574075607466